src/client/java/net/neoforged/neoforge/client/network/ClientPacketDistributor.java

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright (c) NeoForged and contributors
+ * SPDX-License-Identifier: LGPL-2.1-only
+ */
+
+package net.neoforged.neoforge.client.network;
+
+import java.util.Objects;
+import net.minecraft.client.Minecraft;
+import net.minecraft.client.multiplayer.ClientPacketListener;
+import net.minecraft.network.protocol.common.custom.CustomPacketPayload;
+
+/**
+ * Means to distribute serverbound packets
+ */
+public final class ClientPacketDistributor {
+    private ClientPacketDistributor() {}
+
+    /**
+     * Send the given payload(s) to the server
+     */
+    public static void sendToServer(CustomPacketPayload payload, CustomPacketPayload... payloads) {
```

Comment:
I think it might be worth null-checking `payload` and the values of `payloads` here. I looked into `ClientPacketListener#send` and there's no validation that the packet is non-null anywhere, and it eventually reaches netty (which... I presume null-checks it, but is prone to throwing obtuse errors)